### PR TITLE
More items per row addon

### PR DESCRIPTION
**Pull request by mxmou**
See https://github.com/ScratchAddons/ScratchAddons/pull/2947#issuecomment-877235781

### Changes

Adds an addon that allows customizing the number of items shown in a single row of a grid.

### CODE REVIEW

File: addons/items-per-row/classes.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    setClasses("studio-curators", addon.settings.get("studioCurators"), { min: [4, 5], max: [2] });
+    setClasses("projects", addon.settings.get("projects"), { min: [6, 7], max: [4, 3] });
+    setClasses("studios", addon.settings.get("studios"), { min: [6, 7], max: [3] });
+    setClasses("users", addon.settings.get("users"), { min: [12, 15], max: [] });
+  };
+  updateClasses();
+  addon.settings.addEventListener("change", () => {
+    updateClasses();
+    // Handle scratchr2's lazy image loading
+    if (addon.tab.clientVersion === "scratchr2") window.dispatchEvent(new CustomEvent("scroll"));
```
Comment (WorldLanguages) [inline addons/items-per-row/classes.js:28]: Why `CustomEvent` instead of `Event`?

**Comment by WorldLanguages** [comment]
Infinite scrolling works inconsistently with 6 curators per row

**Comment by WorldLanguages** [comment]
A bit offtopic but I would also like to see a reset button for number settings. Would be useful for this addon, as well as the recently added `default-project` addon. For that addon, it would be great because it has presets, and here, it would let you reset a single setting without affecting all of them.

**Comment by WorldLanguages** [comment]
Do you think we should add a notice to `old-studio-layout` hinting users to set studio projects per row to 4 through this addon?

File: addons/items-per-row/classes.js (hunk 1 above)
Comment (mxmou) [reply addons/items-per-row/classes.js:28]: There isn't really a difference in this case, but I don't think `Event` is meant to be initialized directly.

**Comment by mxmou** [comment]
> Infinite scrolling works inconsistently with 6 curators per row

Is limiting the number of curators per row to 5 a good solution? The current maximum makes the cards really small anyway.

> A bit offtopic but I would also like to see a reset button for number settings. Would be useful for this addon, as well as the recently added `default-project` addon. For that addon, it would be great because it has presets, and here, it would let you reset a single setting without affecting all of them.

Reset buttons would probably be useful for all types of settings other than select and Boolean.

> Do you think we should add a notice to `old-studio-layout` hinting users to set studio projects per row to 4 through this addon?

Probably.

File: addons/items-per-row/classes.js (hunk 1 above)
Comment (apple502j) [reply addons/items-per-row/classes.js:28]: `Event` can be constructed so we should use that

**Comment by WorldLanguages** [comment]
> Is limiting the number of curators per row to 5 a good solution? The current maximum makes the cards really small anyway.

Hmm, it also works inconsistently for me with 4 or 5 curators per row.

**Comment by WorldLanguages** [comment]
The problem is caused by #2980, which itself fixed a bug
We could decrease the number of pixels from 500 to, for example, 250, and it would work with up to 4 curators per row.
Or we could rethink the logic being careful not to cause the bug that same PR fixed.

**Comment by WorldLanguages** [comment]
The addon itself is fine, so I'm merging and opening an issue for the infinite-scroll "bug"

**Pull request merged**